### ISSUE TITLE
Export raster as 4-bit-thematic integer?

### ISSUE THREAD

**Issue by kesslerf** [corteva/geocube#27]
Hey Guys,
I am producing a lot of GeoTIFFs in areas with lakes. On my lakes, the raster-value would be either 1 or 0, depending on the state and where no lake is the raster should have NaNs. So far it is working but it takes very long to import the GeoTIFFs in e.g. ArcMap. 
My boss asked me if it is possible to produce 4-bit thematic integers as output.
I did not find any solution for this in the docs, so I just wanted to ask here. 

Thanks a lot,
Felix

**Comment by snowman2** [comment]
@kesslerf I would recommend using the `uint8` dtype and use `3` as `nodata` with `xds = xds.fillna(3).astype("uint8")`:

```python
>>> import xarray
>>> import rioxarray
>>> xds = xarray.DataArray([0
KeyboardInterrupt
>>> import numpy
>>> xds = xarray.DataArray([0, 1, numpy.nan])
>>> xds
<xarray.DataArray (dim_0: 3)>
array([ 0.,  1., nan])
Dimensions without coordinates: dim_0
>>> xds.fillna(3).astype("uint8")
<xarray.DataArray (dim_0: 3)>
array([0, 1, 3], dtype=uint8)
Dimensions without coordinates: dim_0
```
Otherwise, it will change nan to 0:
```python
>>> xds.astype("uint8")
<xarray.DataArray (dim_0: 3)>
array([0, 1, 0], dtype=uint8)
Dimensions without coordinates: dim_0
```

Next, set the `nodata` value in the raster and write to disk:
```python
>>> xds.rio.write_nodata(3, inplace=True)
>>> xds.rio.to_raster(...)
```

**Comment by kesslerf** [comment]
I probably should have mentioned that I am using the Geocube-command on a DataFrame-Object (Shapefile I edit inside my code). I will have a look if I can adapt your solution, thank you very much!

**Comment by snowman2** [comment]
I am going to close this as I assume this has been resolved. Feel free to re-open if you have further questions.

**Comment by byander** [comment]
I don't understand how write de raster like int16.
For my example, I load layer from Geopackage

`soil_df = gpd.read_file(GEOPACKAGE_FILE, layer='solo')`

The layer has a column called 'atp', integer, with values range 0 to 3.

To rasterize, I use this:
`out_grid = make_geocube(vector_data=soil_df, resolution=(10, -10), measurements=['atb'])`
`out_grid['atb'].rio.to_raster(GEOTIFF_CLASS, compress='lzw')`

But the results is a raster float64. I would like that the raster was Int16.  I tried this:
`out_grid['atb'].rio.to_raster(GEOTIFF_CLASS, compress='lzw', dtype=rasterio.uint16)`

but I havn't success. I have this error: 
`ValueError: Given nodata value, nan, is beyond the valid range of its data type, uint16.`

I rasterized with GDAL command with same layer and the results raster is  Int16 type.
